src/drivers/cmakeDriver.ts

```diff
@@ -1714,9 +1764,19 @@ export abstract class CMakeDriver implements vscode.Disposable {
         } else {
             buildPreset.__targets = buildPreset.targets;
         }
-        const args = preset.buildArgs(buildPreset);
+        let args = preset.buildArgs(buildPreset, this.config.buildArgs);
+        args = args.concat('--', ...this.config.buildToolArgs);
```

Comment:
in general, how the user provided arguments are going to be separated in strings? e.g. if the user provides "-G Ninja -DVAR=1 -DANOTHERVAR="value" -- -j 2", how this single string is going to be split in the proper arguments for the process being spawn? Or is it up to the user to provide the arguments already properly split?

Reply:
We are leaving it up to the user to provide the arguments already properly split. However, thanks for calling this out, because I'm noticing that I didn't change this code correctly, making a fix to ensure that we only add `--` once before the buildTool arguments.